internal/serve/httphandler/health_handler.go

```diff
@@ -56,7 +55,7 @@ func (h HealthHandler) ServeHTTP(w http.ResponseWriter, r *http.Request) {
 		"database": dbStatus,
 	}
 
-	if h.EventBrokerType == events.KafkaEventBrokerType {
+	if h.Producer.BrokerType() == events.KafkaEventBrokerType {
```

Comment:
Is it possible for the Producer to be nil in here? Or would we get a NoOp producer when the broker type is set to None?

Better safe than sorry:
```suggestion
	if h.Producer != nil && h.Producer.BrokerType() == events.KafkaEventBrokerType {
```

Reply:
Addressed here 1a934111be946505e433cf298bea500d1b53a3c7